plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -408,6 +408,16 @@ private Object coerce(String context, Type type, Object o) throws Exception {
             return Result.fromString((String)o);
         } else if (o instanceof String && (erased == char.class || erased == Character.class) && ((String) o).length() == 1) {
             return ((String) o).charAt(0);
+        } else if (o instanceof String && (erased == int.class || erased == Integer.class)) {
+            return Integer.valueOf((String)o);
```

Comment:
Is there any issue with changing what previously would have been a `ClassCastException` in some situations into a `NumberFormatException`? It shouldn't be a problem if callers are just handling `Exception`, but anyone looking for specific exceptions might run into issues.

Reply:
Good point, I've done that now.